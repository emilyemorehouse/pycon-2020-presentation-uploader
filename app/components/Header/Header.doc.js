import { describe, PropTypes } from 'react-desc'

export const doc = Header => {
  const DocumentedHeader = describe(Header)
    .description('A Header')
    .usage(
      `import { Header } from 'components/Header';
<Header />`,
    )

  DocumentedHeader.propTypes = {
    id: PropTypes.string
      .description('The DOM id attribute value to use for the element.')
      .defaultValue(false),
  }

  return DocumentedHeader
}
