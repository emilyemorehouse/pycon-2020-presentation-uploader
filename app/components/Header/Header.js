/* eslint-disable react/prop-types */
import React, { useContext } from 'react'
import { observer } from 'mobx-react'
import { User } from 'grommet-icons'

// Components
import { Header as GrommetHeader } from 'grommet'
import { Anchor } from '../Anchor'
import { Box } from '../Box'
import { Menu } from '../Menu'

// Helpers
import { doc } from './Header.doc'
import { createWithDoc } from '../../utils/helpers'

// Stores
import { RootStoreContext } from '../../stores/RootStore'

/**
 *
 * Header
 *
 */
const Header = observer(({ ...rest }) => {
  const { clearStore, user } = useContext(RootStoreContext)

  return (
    <GrommetHeader background="light-2" pad="small" {...rest} style={{ width: '100vw' }}>
      <Box>
        <Anchor href="/" size="large">
          LiveRider
        </Anchor>
      </Box>

      {user.isAuthenticated && (
        <Menu
          dropProps={{ align: { top: 'bottom', left: 'left' } }}
          label={
            <Box pad="8px" background="brand" round>
              <User size="16px" />
            </Box>
          }
          items={[
            { label: 'Settings', onClick: () => {} },
            {
              label: 'Logout',
              onClick: () => {
                clearStore()
              },
            },
          ]}
        />
      )}
    </GrommetHeader>
  )
})
export default createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: Header,
})
