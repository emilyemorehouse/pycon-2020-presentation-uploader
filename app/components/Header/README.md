<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## Header
A Header

## Usage

```javascript
import { Header } from 'components/Header';
<Header />
```

## Properties

**id**

The DOM id attribute value to use for the element.

```
string
```
  