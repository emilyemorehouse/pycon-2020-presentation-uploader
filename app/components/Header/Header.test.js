import React from 'react'
import { render } from '@testing-library/react'
// import '@testing-library/jest-dom/extend-expect' // add some helpful assertions

import Header from './Header'

const renderComponent = (props = {}) => render(<Header {...props} />)

/**
 *
 * Tests for Header
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */
describe('Header', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
