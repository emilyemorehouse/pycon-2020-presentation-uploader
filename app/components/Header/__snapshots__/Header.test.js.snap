// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders and matches snapshot 1`] = `
<div>
  <header
    class="StyledBox-sc-13pk1d4-0 dfQQGl"
    style="width: 100vw;"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jJdKAu"
    >
      <a
        class="StyledAnchor-sc-1rp7lwl-0 eRnWpc"
        href="/"
      >
        LiveRider
      </a>
    </div>
  </header>
</div>
`;
